### PR TITLE
params: Make randao required fork

### PR DESCRIPTION
**Pull request by Mdaiki0730**
## Proposed changes

Change the Randao hard fork to required between Cancun and Kaia.
Randao is now required because the gasless and auction modules released in v2.0.0 and v2.1.0 use a `registry`.
- gasless: https://github.com/kaiachain/kips/blob/main/KIPs/kip-247.md
- auction: https://github.com/kaiachain/kips/blob/main/KIPs/kip-249.md

It also includes a fix for some tests that set randao as optional to nil.

## Types of changes

- [ ] 🐛 Bug fix
- [ ] ✨ Non-hardfork changes (node upgrade not required)
- [ ] 💥 Hardfork / consensus-breaking changes
- [ ] 🧪 Test improvements
- [ ] 🧰 CI / build tool
- [x] ♻️ Chore / Refactor / Non-functional changes

## Checklist

- [x] 📖 I have read the [CONTRIBUTING GUIDELINES](https://github.com/kaiachain/kaia/blob/main/CONTRIBUTING.md) doc
- [x] 📝 I have signed in the PR comment `I have read the CLA Document and I hereby sign the CLA` in first time contribute after having read [CLA](https://gist.github.com/kaiachain-dev/bbf65cc330275c057463c4c94ce787a6)
- [x] 🟢 Lint and unit tests pass locally with my changes (`$ make test`)

### CODE REVIEW

File: params/config.go
Comment (blukat29) [inline params/config.go]:
```
// Optional forks
```

File: params/config.go [hunk 1]
```diff
@@ -578,9 +586,8 @@ func (c *ChainConfig) CheckCompatible(newcfg *ChainConfig, height uint64) *Confi
 // to guarantee that forks can be implemented in a different order than on official networks
```
Comment (blukat29) [inline params/config.go]:
```
// Because some forks (namely KIP103 and KIP160) can be situated at any position, they are not checked for fork order.
```

File: params/config.go
[the remark this answers is not in the document]
Comment (Mdaiki0730) [reply params/config.go]: Reordered at [bd954cf](https://github.com/kaiachain/kaia/pull/701/commits/bd954cff593bcc633499f61d49e727897a556f41). Thanks!

**Review by ian0371 — APPROVED** [review]

**Review by hyeonLewis — APPROVED** [review]
